### PR TITLE
Implement mpi distributed pipeline [draft]

### PR DESCRIPTION
**Pull request by jfomhover**

### CODE REVIEW

File: pipelines/azureml/pipelines/lightgbm_distributed.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
+"""
+Runs an e2e training of LightGBM, then runs inferencing with 3 different versions.
```
Comment (ulfk-ms) [inline pipelines/azureml/pipelines/lightgbm_distributed.py:2]: Update description

File: src/common/lightgbm.py [hunk 2]
```diff
@@ -29,9 +29,12 @@ def callback(self, env: lightgbm.callback.CallbackEnv) -> None:
 
         # loop on all the evaluation results tuples
         for data_name, eval_name, result, _ in env.evaluation_result_list:
+            metric_key = f"{data_name}.{eval_name}"
+            if self.metrics_prefix:
+                metric_key = f"{self.metrics_prefix}.{metric_key}"
```
Comment (ulfk-ms) [inline src/common/lightgbm.py:34]: Have we considered prefix vs postfix? Postfix might yield a more easily usable/scannable sorting in some scenarios.

File: src/common/io.py [hunk 3]
```diff
@@ -27,3 +27,69 @@ def input_file_path(path):
     
     logging.getLogger(__name__).critical(f"Provided INPUT path {path} is neither a directory or a file???")
     return path
+
+
+class PartitioningEngine():
+    """ Class handles partitioning files into chunks with various strategies. """
```
Comment (ulfk-ms) [inline src/common/io.py:33]: Maybe add here that this is line-based, i.e. groups (that need to be processed together) aren't allowed

**Review by ulfk-ms — APPROVED** [review]

File: src/common/lightgbm.py (hunk 2 above)
Comment (jfomhover) [reply src/common/lightgbm.py:33]: Discussed in the other pr #61 and filed as issue #63

File: src/common/io.py (hunk 3 above)
Comment (jfomhover) [reply src/common/io.py]: Note: this change has been pushed as part of another pr #58

**Pull request merged**